apps/src/templates/teacherDashboard/teacherSectionsRedux.js

```diff
@@ -610,6 +610,25 @@ export function isRosterDialogOpen(state) {
   return getRoot(state).isRosterDialogOpen;
 }
 
+export function oauthProvider(state) {
+  return getRoot(state).provider;
+}
+
+export function sectionCode(state, sectionId) {
+  return (getRoot(state).sections[sectionId] || {}).code;
+}
+
+export function sectionProvider(state, sectionId) {
+  if (isSectionProviderManaged(state, sectionId)) {
+    return oauthProvider(state);
+  }
+  return null;
+}
+
+export function isSectionProviderManaged(state, sectionId) {
+  return !!(getRoot(state).sections[sectionId] || {}).providerManaged;
+}
+
```

Comment:
I started including a `getRoot` method after I saw Paul do this, I _think_ because it was a step closer to his ideal of a dux module that could be mounted anywhere on the state and still be usable.

I've stuck with this because if we _must_ have knowledge of the state layout, I like having that knowledge the smallest number of places, and once in the dux module seems like a smaller footprint than in each consumer component.